src/Traits/AdapterHelper.php

```diff
@@ -12,25 +12,25 @@
 
 trait AdapterHelper
 {
+    protected PropertyGroupOptionEntity $propertyGroupOptionEntity;
+    protected PluginConfig $pluginConfig;
+
     /**
      * @return Attribute[]
      */
-    public static function getPropertyGroupOptionAttributes(
+    protected function getPropertyGroupOptionAttributes(
         PropertyGroupOptionCollection $collection,
-        PluginConfig $pluginConfig,
+        PluginConfig $pluginConfig
     ): array {
+        $this->pluginConfig = $pluginConfig;
         $attributes = [];
 
         foreach ($collection as $propertyGroupOptionEntity) {
-            $group = $propertyGroupOptionEntity->group;
-            if ($group && !$group->filterable) {
+            if ($propertyGroupOptionEntity->group && !$propertyGroupOptionEntity->group->filterable) {
                 continue;
             }
-
-            $attributes = array_merge($attributes, static::getAttributePropertyAsAttribute(
-                $propertyGroupOptionEntity,
-                $pluginConfig,
-            ));
+            $this->propertyGroupOptionEntity = $propertyGroupOptionEntity;
```

Comment:
Why do you need to save this in a member variable?
There should not be a lot of code changes needed, if you move the functions. The logic is exactly the same, but just shared by 2 adapters.